src/main/java/io/stargate/sgv2/jsonapi/service/operation/MetadataAttempt.java

```diff
@@ -0,0 +1,275 @@
+      return new ComplexTypes.VectorType(PrimitiveTypes.FLOAT, vt.getDimensions(), vectorizeConfig);
+    } else if (columnMetadata.getType() instanceof MapType mt) {
+      if (!mt.isFrozen()) {
+        final Optional<ApiDataTypeDef> apiDataTypeDefKey = ApiDataTypeDefs.from(mt.getKeyType());
+        final Optional<ApiDataTypeDef> apiDataTypeDefValue =
+            ApiDataTypeDefs.from(mt.getValueType());
+        if (apiDataTypeDefKey.isPresent() && apiDataTypeDefValue.isPresent()) {
+          return new ComplexTypes.MapType(
+              PrimitiveTypes.fromString(apiDataTypeDefKey.get().getApiType().getApiName()),
+              PrimitiveTypes.fromString(apiDataTypeDefValue.get().getApiType().getApiName()));
+        }
+      }
+      // return unsupported format
+      return new ComplexTypes.UnsupportedType(mt.asCql(true, false));
+
+    } else if (columnMetadata.getType()
+        instanceof com.datastax.oss.driver.api.core.type.ListType lt) {
+      if (!lt.isFrozen()) {
+        final Optional<ApiDataTypeDef> apiDataTypeDef = ApiDataTypeDefs.from(lt.getElementType());
+        if (apiDataTypeDef.isPresent()) {
+          return new ComplexTypes.ListType(
+              PrimitiveTypes.fromString(apiDataTypeDef.get().getApiType().getApiName()));
+        }
+      }
+      // return unsupported format
+      return new ComplexTypes.UnsupportedType(lt.asCql(true, false));
+
+    } else if (columnMetadata.getType()
+        instanceof com.datastax.oss.driver.api.core.type.SetType st) {
+      if (!st.isFrozen()) {
+        final Optional<ApiDataTypeDef> apiDataTypeDef = ApiDataTypeDefs.from(st.getElementType());
+        if (apiDataTypeDef.isPresent()) {
+          return new ComplexTypes.SetType(
+              PrimitiveTypes.fromString(apiDataTypeDef.get().getApiType().getApiName()));
+        }
+      }
+      // return unsupported format
+      return new ComplexTypes.UnsupportedType(st.asCql(true, false));
+    } else {
+      final Optional<ApiDataTypeDef> apiDataTypeDef =
+          ApiDataTypeDefs.from(columnMetadata.getType());
+      if (apiDataTypeDef.isPresent())
+        return PrimitiveTypes.fromString(apiDataTypeDef.get().getApiType().getApiName());
+      else {
+        // Need to return unsupported type
+        return new ComplexTypes.UnsupportedType(columnMetadata.getType().asCql(true, false));
+      }
+    }
+  }
+
+  /**
+   * Object used to build the response for listTables command
+   *
+   * @param name
+   * @param definition
+   */
+  @JsonPropertyOrder({"name", "definition"})
+  @JsonInclude(JsonInclude.Include.NON_NULL)
+  public record TableResponse(String name, TableDefinition definition) {
+
+    @JsonPropertyOrder({"columns", "primaryKey"})
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    record TableDefinition(Map<String, ColumnType> columns, PrimaryKey primaryKey) {}
+  }
+
+  protected List<TableSchemaObject> getTables() {
+    return keyspaceMetadata
+        .get()
```

Comment:
nit: lint checking kicks in to say this is access to an Optional without checking if present, we should unwrap the optional in executeStatement and reference the KeyspaceMetadata rather than use Optional<KeyspaceMetadata>